time.go

```diff
@@ -227,6 +228,15 @@ func (t DateTime) MarshalBSONValue() (bsontype.Type, []byte, error) {
 // assumed to be valid. UnmarshalBSONValue must copy the BSON value bytes if it
 // wishes to retain the data after returning.
 func (t *DateTime) UnmarshalBSONValue(tpe bsontype.Type, data []byte) error {
+	if tpe == bsontype.Null {
+		*t = DateTime{}
+		return nil
+	}
+
+	if len(data) != 8 {
```

Comment:
Why it has to be 8 bytes?

Reply:
because otherwise binary.LittleEndian.Uint64(data) throws a panic.